src/ingest-pipeline/airflow/dags/catch_pipeline_errors.py

```diff
@@ -0,0 +1,163 @@
+import json
+from pprint import pprint
+from requests.exceptions import HTTPError
+from requests import codes
+import traceback
+
+from airflow.providers.http.hooks.http import HttpHook
+from airflow.utils.email import send_email
+
+from utils import get_auth_tok
+
+"""
+TODO
+- this requires crypt_auth_tok to be an op_kwarg, is that reasonable?
+- fix DRY issues (e.g. get_submission_context?, set_status)
+- type hinting
+"""
+
+
+class FailureCallbackException(Exception):
+    pass
+
+
+class FailureCallback:
+    """
+    List should be overwritten by each subclass with appropriate values.
+    """
+
+    external_exceptions = []
```

Comment:
Fine for first iteration.

Might want to think about making this more of a mapping of exceptions to a list of "reporting" functions

Reply:
Do you mean like a list of notifications that should  be sent per specific error type, in the case where we know certain errors should be handled in specific ways? e.g.:
PreflightError: [send_failure_email, send_asana_notification],
ValidatorError: [send_failure_email]

My thought here is that perhaps it would make sense to separate internal and external email notifications into two different methods. Haven't totally thought that through yet. At the moment all errors generate an internal email at a minimum. That should be easy enough to keep doing unless no email is a better default.